backend/trivia_api/views.py

```diff
@@ -61,6 +61,71 @@ def users(user_id):
     return dummy_users
 
 
+@blueprint.route("/signup", methods=["GET", "POST"])
+def signup_post():
+    if request.method == "POST":
+        # return jsonify({"message": "success"}, 200)
+        data = request.get_json("password")
+        newusers = data["username"]
+        #  newusers2 = str(newusers)
+        newpasswords = data["password"]
+        passwords = generate_password_hash(newpasswords, method="sha256")
+        newuser = User(username=newusers)
+        newpass = User(password=passwords)
+        if (
+            User.query.filter_by(username=newuser.username).first() is None
+            and User.query.filter_by(password=newpass.password).first() is None
+        ):
+            entry = User(username=newuser.username, password=newpass.password)
+            db.session.add(entry)
+            db.session.commit()
+            return jsonify({"code": 200, "message": "failed", "message": "success"})
+        else:
+            return jsonify(
+                {
+                    "code": 400,
+                    "message": "failed",
+                    "message": "User Name is already Taken",
+                }
+            )
+    #  return jsonify({"new": newusers})
+    return jsonify({"code": 400, "message": "failed"})  # what am I returning here
+
+
+@login_manager.user_loader
+def loaduser(user_id):
+    return User.query.get(user_id)
+
+
+@blueprint.route("/login", methods=["GET", "POST"])
+def login_post():
+    username = request.get_json("username")
+    checkusers = username["username"]
+    password = username["password"]
+    #     remember = bool(request.form.get("remember"))
+    user = User.query.filter_by(username=checkusers).first()
+    if not user or not check_password_hash(user.password, password):
+        return jsonify(
+            {
+                "code": 400,
+                "message": "failed",
+                "message": "You have entered the wrong login! Try again",
+            }
+        )
+    #         flash("Please check your login details and try again.")
+    #         return redirect(url_for("blueprint.login"))
+    else:
+        login_user(user)
+        return jsonify({"code": 200, "message": "failed", "message": "success"})
```

Comment:
```suggestion
    login_user(user)
    return {"success": True}
```

else is unnecessary because of return from the if